src/service/resources-impl.js

```diff
@@ -1359,13 +1362,13 @@ export class Resource {
      * Pending change height that was requested but could not be satisfied.
      * @private {number|undefined}
      */
-    this.pendingChangeHeight_;
-
-    /** @private {?function(*)} */
-    this.whenFirstLayoutCompleteResolve_ = null;
+    this.pendingChangeHeight_ = undefined;
 
-    /** @private @const {?Promise} */
-    this.whenFirstLayoutCompletePromise_ = null;
+    /** @private @const {!Promise} */
+    this.loadPromise_ = new Promise(resolve => {
```

Comment:
so @dvoytenko requested we put the initialization of these promises in `loaded` so that we lazily create the promise for a `resource` if nothing is monitoring it.